pkg/curatedpackages/packageinstaller.go

```diff
@@ -47,9 +56,27 @@ func newPackageController(installer ChartInstaller, runner KubectlRunner, spec *
 	return NewPackageControllerClient(installer, runner, kubeConfig, imageUrl, chart.Name, chart.Tag())
 }
 
+func newCertManager(runner KubectlRunner, spec *cluster.Spec) *CertManagerClient {
```

Comment:
Instead of creating this here, can we pass it in to the `NewInstaller` constructor above? This will do dependency injection and make it easier to unit test. Just trying to think of the simplest way to create dependencies, unless we have been doing this elsewhere.

Reply:
Similar to my response in the other comment, we currently need tailored user message `PrintCertManagerDoesNotExistMsg` to guide users, and I'm trying to mirror how we handle package controller here to handle cert managers. I have added `TestPackageInstallerFailWhenCertManagerFails` unit test in `packageinstaller_test.go` to test out this scenario along with functional testing. Let me know if this works!